### ISSUE TITLE
TList next

### ISSUE THREAD

**Issue by cachoJDG** [marcelogarberoglio/PI#253]
![image](https://github.com/marcelogarberoglio/PI/assets/130168865/ed4a501b-fb7e-41a1-948e-2628e74c950d)
buenas profe no me queda claro cuando nos conviene incorporar al CDT un next, al igual que si debería llamar la función de liberar dos veces para liberar tanto el first como el next o todo de una.

**Comment by marcelogarberoglio** [comment]
El next lo agregamos para poder hacer las funciones toBegin, hasNext y next. Es un campo auxiliar para que el usuario pueda recorrer toda la lista en forma eficiente, sin usar el get
Al liberar se liberan los nodos, así que no, no hay que llamar dos veces, por ejemplo en este caso

int * p = malloc(120);
int * q = p;

¿Cuáles de las opciones son correctas?

Opción 1: free(p); free(q);
Opción 2: free(p);
Opción 3: free(q);

**Comment by cachoJDG** [comment]
la 2 y la 3 porque cualquiera de las dos esta apuntando a lo mismo asi q libera el mismo espacio de memoria.

**Comment by marcelogarberoglio** [comment]
Exacto, lo mismo pasaría si next quedó apuntando al primero, por eso no hay que modificar el freeList.
